packages/space-opera/src/components/mobile_view/open_mobile_view.ts

```diff
@@ -0,0 +1,350 @@
+  // dynamically send certain GET requests
+  getUpdatedContent() {
+    return {
+      gltfChanged: this.isNewModel(), stateChanged: this.stateHasChanged(),
+          envChanged: this.isNewSource(this.urls.env, this.lastUrlsSent.env),
+          envIsHdr: this.envIsHdr
+    }
+  }
+
+  async postObject(obj: Object, url: string) {
+    await fetch(url, {
+      method: 'POST',
+      body: JSON.stringify(obj),
+    })
+        .then(response => {
+          console.log('Success:', response);
+        })
+        .catch((error) => {
+          console.log('Error:', error);
+          throw new Error(`Failed to post: ${url}`);
+        });
+  }
+
+  // Send any state, model, or image that has been updated since the last update
+  async postInfo() {
+    this.isSendingData = true;
+    const updatedContent = this.getUpdatedContent();
+    await this.postObject(updatedContent, this.updatesPipeUrl)
+
+    if (updatedContent.gltfChanged) {
+      await this.sendBlob(this.urls.gltf!, 'gltf');
+    }
+
+    if (updatedContent.stateChanged) {
+      await this.postObject(this.snippet, this.snippetPipeUrl);
+      this.lastSnippetSent = {...this.snippet};
+    }
+
+    if (updatedContent.envChanged) {
+      await this.sendBlob(this.urls.env!, 'env');
+    }
+
+    this.contentHasChanged = this.getHasChanged();
+    this.isSendingData = false;
+  }
+
+  async waitForPing() {
+    await fetch(this.mobilePingUrl)
+        .then(response => response.json())
+        .then(responseJson => {
+          if (responseJson.isPing) {
+            this.haveReceivedResponse = true;
+          }
+        })
+        .catch((error) => {
+          console.error('Error:', error);
+        });
+  }
+
+  // Opens the modal that displays the QR Code
+  openModal() {
+    this.mobileModal.open();
+  }
+
+  // Called each time the editor hasn't received a ping from mobile
+  async onDeploy() {
+    this.openModal();
+    await this.waitForPing();
+    if (this.haveReceivedResponse) {
+      this.postInfo();
+    } else {
+      this.onDeploy();
+    }
+  }
+
+  // Initialize AR values and start deploy loop
+  async onInitialDeploy() {
+    this.isDeployed = true;
+    reduxStore.dispatch(dispatchAr(true));
+    reduxStore.dispatch(dispatchArModes('webxr scene-viewer quick-look'));
+    await this.onDeploy();
+  }
+
+  renderDeployButton() {
+    return html`
+    <mwc-button unelevated
+      icon="file_download"
+      ?disabled=${!this.isDeployable}
+      @click=${this.onInitialDeploy}>
+        Deploy Mobile
+    </mwc-button>`
+  }
+
+  onEnableARChange() {
+    reduxStore.dispatch(dispatchAr(this.arCheckbox.checked));
+  }
+
+  onSelectArMode(event: CustomEvent) {
+    const dropdown = event.target as Dropdown;
+    const key = dropdown.selectedItem?.getAttribute('value') || undefined;
+    if (key === 'default' || key === 'webxr') {
+      reduxStore.dispatch(dispatchArModes('webxr scene-viewer quick-look'));
```

Comment:
Let's simplify this. Maybe just a checkbox to default to scene-viewer instead of webxr? Quicklook can be always enabled, since it's the only option on iphones, and it's really controlled by the presence of an ios-src.